### PR TITLE
ci(pie-monorepo): Set up PR previews using AWS Amplify

### PR DESCRIPTION
**Pull request by xander-marjoram**
## Author Checklist (complete before requesting a review)
- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

**Comment by xander-marjoram** [comment]
Closing to test PR preview deletion

**Comment by xander-marjoram** [comment]
Testing again

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -161,32 +200,67 @@ jobs:
...
+      - name: Deploy Main (Amplify)
+        if: ${{ github.ref == 'refs/heads/main' }}
+        uses: ./.github/actions/amplify-deploy
+        with:
+          amplify-app-id: ${{ env.DOCS_AMPLIFY_ID }}
+          aws-region: 'eu-west-1'
+          aws-access-key-id: ${{ secrets.AWS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          branch-name: 'main'
+          bucket-name: ${{ secrets.AWS_STORYBOOK_MAIN_BUCKET }}
```
Comment (xander-marjoram) [inline .github/workflows/build.yml:262]:
```suggestion
          bucket-name: ${{ secrets.AWS_DOCS_MAIN_BUCKET }}
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -161,32 +200,67 @@ jobs:
...
+          package-name: 'pie-docs'
```
Comment (xander-marjoram) [inline .github/workflows/build.yml:263]:
```suggestion
          amplify-app-id: ${{ env.STORYBOOK_AMPLIFY_ID }}
          aws-region: 'eu-west-1'
          aws-access-key-id: ${{ secrets.AWS_KEY_ID }}
          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
          branch-name: 'main'
          bucket-name: ${{ secrets.AWS_STORYBOOK_MAIN_BUCKET }}
          package-name: 'pie-storybook'
```

**Review by jamieomaguire — APPROVED** [review]

**Review by ashleynolan — APPROVED** [review]

**Pull request merged**